cooperator/models/cooperative_membership.py

```diff
@@ -243,3 +243,80 @@ def get_share_quantities(self):
             total_shares[line.share_product_id.id] += line.share_number
 
         return total_shares
+
+    def assign_membership(self):
+        """Make sure the following things are in order for all memberships in
+        the recordset:
+
+        - ``member`` is set to True and ``old_member`` is set to False.
+        - A cooperator register number is assigned if one did not yet exist.
+        - A user is created if one did not yet exist.
+
+        This should all be idempotent.
+        """
+        self.write({"member": True, "old_member": False})
+        self.assign_new_cooperator_register_number()
+        for membership in self:
+            # TODO: Should this test be inside of create_user?
+            if membership.company_id.create_user:
+                membership.create_user()
+
+    def assign_new_cooperator_register_number(self):
+        """Set a new cooperator register number on the memberships if one is not
+        already assigned.
+        """
+        for membership in self:
+            # Already exists; nothing to do.
+            if membership.cooperator_register_number:
+                continue
+            membership.cooperator_register_number = (
+                self.company_id.get_next_cooperator_number()
+            )
+
+    def create_user(self):
+        """Create a new user for the memberships if one does not already exist.
+
+        If a user exists, but its 'active' field is False, then the user is
+        reactivated and the membership's company_id becomes the user's only
+        company_id.
+
+        If a new partner is created, an e-mail is sent to them with instructions
+        on how to set their passport.
+        """
+        user_obj = self.env["res.users"].sudo()
+
+        for membership in self:
+            partner = membership.partner_id
+            email = partner.email
+
+            user = user_obj.with_context(active_test=False).search(
+                [("login", "=", email)]
+            )
+            if user:
```

Comment:
the internal tasks suggests to test against `partner.user_ids` instead of searching a user by email address. what is the status of this regarding these changes?

Reply:
Made a commit for this, plus wrote tests.